src/chatkit.ts

```diff
@@ -600,6 +611,88 @@ export default class Chatkit {
     }).then(() => {})
   }
 
+  sendMessage(options: SendMessageOptions, jwtToken: TokenWithExpiry | null = null, autoJwt: boolean = false): Promise<any> {
+    if (typeof options.senderId === typeof undefined) {
+        throw new Error('You must provide the ID of the user that you want to set as the sender of the message');
+    }
+    if (typeof options.roomId === typeof undefined) {
+        throw new Error('You must provide the ID of the room that you want to add the message to');
+    }
+    if (typeof options.text === typeof undefined) {
+        throw new Error('You must provide some text for the message');
+    }
+
+    interface Body {
+      text: string;
+      attachment?: {
+        resource_link: string;
+        type: 'image' | 'video' | 'audio' | 'file';
+      }
+    }
+
+    const body: Body = {
+      text: options.text
+    };
+
+    if (typeof options.attachment !== typeof undefined && options.attachment) {
+      if (typeof options.attachment.resourceLink === typeof undefined) {
+        throw new Error('You must provide the resource_link for the attachment');
+      }
+      if (typeof options.attachment.type === typeof undefined
+        || !(['image', 'video', 'audio', 'file'].indexOf(options.attachment.type) > -1) ) {
+        throw new Error('You must provide the type for the attachment. This can be one of image, video, audio or file');
+      }
+      if (options.attachment) {
+        body.attachment = {
+          resource_link: options.attachment.resourceLink,
+          type: options.attachment.type
+        };
+      }
+    }
+
+    const requestOptions: RequestOptions = {
+      method: 'POST',
+      path: `/rooms/${options.roomId}/messages`,
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: body
+    };
+
+    if (jwtToken === null && !autoJwt) {
+      throw new Error('Your request can\'t be authenticated. Pass correct JWT token or use auto option.');
+    }
+
+    // If auto is false and token provided
+    if (jwtToken !== null && !autoJwt) {
+      requestOptions.jwt = this.getToken(jwtToken);
+    }
+
+    // Override JWT token if auto set to true
+    if (autoJwt) {
```

Comment:
I don't think we need to split the token concepts into two parts, `autoJwt` and `jwtToken`.

I think the signature of the function should remain:

```js
sendMessage(options: SendMessageOptions)
```

and `SendMessageOptions` should have an optional property called `useSuToken`, which defaults to `false`. If set to true then all that needs to happen is when `generateAccessToken` is called it's called with `su: true` along with the `userId`. Otherwise just call it with the `userId`. 

WDYT?